lib/mssql.js

```diff
@@ -155,18 +155,34 @@ MsSQL.prototype.create = function (model, data, callback) {
   var modelPKID = this.idName(model);
 
   var columns = fieldsAndData.fields ? " (" + fieldsAndData.fields + ")" : "";
-  var sql = "INSERT INTO " + tblName + columns + MsSQL.newline;
+  var sql = '';
+
+  var props = this._models[model].properties;
+  var isIdentity = (props[modelPKID].type === Number &&
+    props[modelPKID].generated !== false);
+
+  if (isIdentity && data[modelPKID] != null) {
+    // insert with identity_insert
+    sql += "SET IDENTITY_INSERT " + tblName + " ON;" + MsSQL.newline;
+  }
+
+  sql += "INSERT INTO " + tblName + columns + MsSQL.newline;
   sql += "OUTPUT INSERTED." + this.columnEscaped(model, modelPKID)
     + " AS insertId" + MsSQL.newline;
   sql += (fieldsAndData.paramPlaceholders ?
     "VALUES (" + fieldsAndData.paramPlaceholders + ");" :
     "DEFAULT VALUES;");
 
+  if (isIdentity && data[modelPKID] != null) {
+    sql += "SET IDENTITY_INSERT " + tblName + " OFF;" + MsSQL.newline;
+  }
```

Comment:
What happens when the `INSERT` command fails, will MySQL continue executing other commands included in the multi-command string? If it does not, then `IDENTITY_INSERT` won't be turned off when `INSERT` failed.